packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -340,10 +340,26 @@ class FlutterPlugin implements Plugin<Project> {
                         "packages", "flutter_tools", "gradle", "src", "main", "kotlin",
                         "dependency_version_checker.gradle.kts")
                 project.apply from: dependencyCheckerPluginPath
-            } catch (Exception ignored) {
+            } catch (Exception e) {
+                // If the exception was thrown by us in the dependency version checker plugin then
```

Comment:
This is a bit gross as any change to that message would cause this unrelated code to fail. 

Can you modify the exception to have a root class maybe and catch that root class and rethrow?

Reply:
name idea: `DependencyValidationException` :P

Reply:
Yeah, I agree it's pretty gross. 

The reason I wrote it this way is that I thought at first that any exception class I wrote for use in the kotlin side (where we throw) wouldn't resolve in the groovy side (where we catch). But I think I'm mistaken about that, it just gets wrapped in a `GradleException`, so I'll still have to dig in like I am here, but won't have to rely on error message at least

Reply:
Ok, this is still gross but hopefully less so.

I've made changes so that we have a `DependencyValidationException` defined in `flutter.groovy`, and then we throw that from within the kotlin file. We still catch broadly, because Gradle wraps our exception in a couple others, but now we test for that error class with `instanceof`.